packages/dd-trace/src/profiling/exporters/agent.js

```diff
@@ -177,9 +178,10 @@ class AgentExporter {
         if (this._url.protocol === 'unix:') {
           options.socketPath = this._url.pathname
         } else {
-          options.protocol = this._url.protocol
-          options.hostname = this._url.hostname
-          options.port = this._url.port
+          const httpOptions = urlToHttpOptions(this._url)
```

Comment:
Saw notes in another file that this library was holding off on using urlToHttpOptions due to version compatibility issues. However with v5 now supporting node 18+, it seems like this should be okay to use now.

Reply:
I see `urlToHttpOptions` is marked as Node 15+; v3 still supports Node 14 so I think `dont-land-on-v3.x` label is warranted – I added it. Ordinarily I'd ask for a runtime check of the form `if (semver.gte(process.version, '16.0.0'))` so we minimize the divergence of v3 to v5, but v3 EOL is 2 months away so I guess we can cope with this not getting backported.